src/core/egglog-herbie.rkt

```diff
@@ -169,7 +179,13 @@
   (for ([(tag schedule-params) (in-dict tag-schedule)])
     (match tag
       ['lifting (set! run-schedule (cons `(saturate lifting) run-schedule))]
-      ['lowering (set! run-schedule (cons `(saturate lowering) run-schedule))]
+      ['lowering
+       (set! run-schedule (cons `(saturate lowering) run-schedule))
+
+       (define const-fold-best-iter-limit
+         (egglog-unsound-detected curr-program 'const-fold schedule-params run-schedule))
+
+       (set! run-schedule (cons `(repeat ,const-fold-best-iter-limit const-fold) run-schedule))]
```

Comment:
My model of constant folding in `egg` is this: after running a ruleset for 1 iteration, saturate using constant folding. Repeat until resource limits hit. Therefore, I would expect constant folding to be in the `else` case of this match expression, not in `lowering` (which is a special ruleset).